requirements_tools/check_requirements.py

```diff
@@ -1,15 +1,14 @@
-from __future__ import absolute_import
-from __future__ import print_function
-from __future__ import unicode_literals
+from __future__ import annotations
 
-import io
 import itertools
 import os.path
 import sys
 from operator import attrgetter
+from typing import Iterable
 
 import pkg_resources
 import pytest
+from pkg_resources import Requirement
```

Comment:
I would probably use the one from `packaging` directly

Reply:
okay, then we'd have to pick up `packaging` as an `install_requires` and then use `packaging.requirements.Requirement` that'll probably make other things and typing easier. I hope that's fine!

Reply:
yeah it should get added -- won't make a difference though as it's already pulled in via pytest

Reply:
ah looks like it's not that simple unfortunately...
pgk_resources' `Requirements` inherits from (the vendored) `packaging.requirements.Requirement`: https://github.com/pypa/setuptools/blob/bfd92b008cbda2c1252eddd2f3e54b91f61969fb/pkg_resources/__init__.py#L3092-L3137

It adds the `parse` classmethod and a few attributes (`key` and `specs`)
I think we have to keep it as is...